ChustaSoft.Tools.DBAccess.MongoDb/Repository.cs

```diff
@@ -1,79 +1,102 @@
-﻿using System;
+﻿using ChustaSoft.Common.Contracts;
+using MongoDB.Driver;
+using System;
 using System.Collections.Generic;
 using System.Linq;
 
 namespace ChustaSoft.Tools.DBAccess
 {
     public class Repository<TEntity, TKey> : RepositoryBase<IMongoContext, TEntity>, IRepository<TEntity, TKey>
-        where TEntity : class
+        where TEntity : class, IKeyable<TKey>
     {
+        private readonly IMongoCollection<TEntity> _dbSet;
 
         public IQueryable<TEntity> Query => throw new NotImplementedException();
 
 
         public Repository(IMongoContext context) 
             : base(context)
-        { }
-
-
+        {
+            _dbSet = context.GetCollection<TEntity>();
+        }
 
         public TEntity GetSingle(TKey id)
         {
-            throw new NotImplementedException();
+            var idFilter = CreateIdFilter(id);
+
+            return _dbSet
+                .Find(idFilter)
+                .FirstOrDefault();
         }
 
         public TEntity GetSingle(Action<ISingleResultSearchParametersBuilder<TEntity>> searchCriteria)
         {
-            throw new NotImplementedException();
+            var searchParams = MongoSearchParametersBuilder<TEntity, MongoSearchParameters<TEntity>>.GetParametersFromCriteria(searchCriteria);
```

Comment:
Isolate this line in a private method and reuse here and inside GetMultiple

Reply:
[Corrected in linked commit](https://github.com/ChustaSoft/DBAccess/pull/22/commits/a41ac9ad2ebb8ecf338b40fb4ae1d55afe2f69ac)